torchmetrics/functional/text/rouge.py

```diff
@@ -101,17 +106,31 @@ def _normalize_and_tokenize_text(text: str, stemmer: Optional[Any] = None) -> Se
             An input sentence.
         stemmer:
             Porter stemmer instance to strip word suffixes to improve matching.
+        normalizer:
+            A user's own normalizer instance.
+            If this is none, `replacing any non-alpha-numeric characters with spaces` is default.
+            This instance must have method named ``normalize``. This method must take a string and return a string.
+        tokenizer:
+            A user's own tokenizer instance. If this is none, `spliting by spaces` is default
+            This instance must have method named ``tokenize``. This method must take a string and return `List[str]`
     """
-    # Replace any non-alpha-numeric characters with spaces.
-    text = re.sub(r"[^a-z0-9]+", " ", text.lower())
+    if normalizer:
+        text = normalizer.normalize(text)
+    else:
+        # Replace any non-alpha-numeric characters with spaces.
+        text = re.sub(r"[^a-z0-9]+", " ", text.lower())
+
+    if tokenizer:
+        tokens = tokenizer.tokenize(text)
+    else:
+        tokens = re.split(r"\s+", text)
```

Comment:
Wondering if a requirement that each normalizer/tokenizer have `normalize`/`tokenize` method respectively is not too strict. Wouldn't be better to have something like
```suggestion
    if normalizer:
        text = normalizer(text)
    else:
        # Replace any non-alpha-numeric characters with spaces.
        text = re.sub(r"[^a-z0-9]+", " ", text.lower())

    if tokenizer:
        tokens = tokenizer(text)
    else:
        tokens = re.split(r"\s+", text)
```
i.e. normalizer/tokenizer is callable? Also, with this implementation, any method can be easily wrapped with a `partial` by a user?
Any thoughts @Borda?

Reply:
that looks nice, but don't have a strong pref...
@SkafteNicki ?

Reply:
I thought so at first, but I thought it was better to receive and process it as an object in optimizing even if it was a little more strict than to create a regular expression object or something in every function.

Reply:
In general, I'm open to both options. Maybe, can you add an example using own tokenizer and normalizer into `tm_examples` as it is the case, for example, for `BERTScore`? I think it can be beneficial for this PR as the usage of own tokenizer and normalizer is not included in the reference `rouge_score` package :]

Reply:
umm... I will try then!